src/commands/deck.ts

```diff
@@ -84,28 +93,80 @@ export class DeckCommand extends Command {
 			side: namedDeck.side.reduce(count, {})
 		};
 		// sum up the number of cards in each section for the headings
-		// TODO: get monster/spell/trap counts
+		// we do this seperately from the some of the below to not assume a card fits in exactly one of the given categories
 		const sum = (acc: number, cur: number): number => acc + cur;
 		const sums = {
 			main: Object.values(deckCounts.main).reduce(sum, 0),
 			extra: Object.values(deckCounts.extra).reduce(sum, 0),
 			side: Object.values(deckCounts.side).reduce(sum, 0)
 		};
+		// count the number of each meaningful card type in the deck
+		const typeCount =
+			(types: string[], field: "type" | "subtype") => (acc: Record<string, number>, val: number) => {
+				const card = cardMemo.get(val);
+				if (card) {
+					for (const type of types) {
+						if (card[field] === type) {
+							acc[type] = acc[type] ? acc[type] + 1 : 1;
+						}
+					}
+				}
+				return acc;
+			};
+		const mainTypes = ["Monster", "Spell", "Trap"];
+		const extraTypes = ["Fusion", "Synchro", "Xyz", "Link"];
+		const typeCounts = {
+			main: [...deck.main].reduce(typeCount(mainTypes, "type"), {}),
+			extra: [...deck.extra].reduce(typeCount(extraTypes, "subtype"), {}),
+			side: [...deck.side].reduce(typeCount(mainTypes, "type"), {})
+		};
 		// print information into embed
 		const printCount = (value: [string, number]): string => `${value[1]} ${value[0]}`;
+		const plurals: Record<string, string> = {
+			Monster: "Monsters",
+			Spell: "Spells",
+			Trap: "Traps",
+			Fusion: "Fusions",
+			Synchro: "Synchros",
+			Xyz: "Xyz",
+			Link: "Links"
+		};
```

Comment:
This is a step up from Emcee, which does always use plural. In general though this is not the way to handle internationalization if we want the labels to also be in other languages.

Reply:
Naturally, discord/discord-api-docs#2313 is dead though :/

Reply:
Naturally. 
> This request is on our internal issue tracker so we won't forget 👍
> msciotti commented on 16 Jan

Native support aside Bastion already has reason to request locale for the user to get a deck in the Yu-Gi-Oh! supported language of their choice, so when multilanguage support is added to commands that parameter will be normalised. At that stage we could externalise a dictionary of translations in some fashion, perhaps?